src/container/srv_epoch.c

```diff
@@ -503,14 +503,18 @@ ds_cont_update_snap_iv(struct cont_svc *svc, uuid_t cont_uuid)
 		goto out_lock;
 	}
 
-	rc = cont_iv_snapshots_update(svc->cs_pool->sp_iv_ns, cont_uuid,
-				      snapshots, snap_count);
-	if (rc != 0)
-		D_ERROR(DF_CONT": Failed to update snapshots IV: %d\n",
-			DP_CONT(svc->cs_pool_uuid, cont_uuid), rc);
-
-	D_FREE(snapshots);
```

Comment:
[Question] If we don't touch the IV "state" (apologies for not knowing the correct IV terminology) before releasing the DB lock, then what will happen if an IV_FETCH request comes in after we release the DB lock but before we update the IV "state"? Will this IV_FETCH see stale IV values?

Reply:
Yes, the IV value on other different target might be stale. But they will be synchronized eventually, if there is need to make them synchronize, there should be other mechanism. Not sure we have such use case for the moment.

IMHO, it is improper to use a simple local lock to implement global synchronization.

Reply:
I'm only considering the IV "state" in the local engine who is the root of the IV.

Is it possible that the RDB TX commits and a local IV fetch before the IV update doesn't see the committed changes? This depends on a couple of IV questions:

- Does the IV update guarantee that it will never yield?
- Does the IV fetch algorithm at the IV root only looks at the IV "state" or does it also checks with RDB? If it looks at the IV "state" only, then we could consider invalidate the IV locally (i.e., no RPCs involved) while holding the DB lock.